elf/include/bpf.h

```diff
@@ -605,4 +605,28 @@ struct xdp_md {
 	__u32 data_end;
 };
 
+#define BUF_SIZE_MAP_NS 256
+
+typedef struct bpf_map_def {
+	unsigned int type;
+	unsigned int key_size;
+	unsigned int value_size;
+	unsigned int max_entries;
+	unsigned int map_flags;
+	unsigned int pinning;
+	char namespace[BUF_SIZE_MAP_NS];
+} bpf_map_def;
+
+typedef struct bpf_map {
```

Comment:
struct `bpf_map_def` and struct `bpf_map` should not be in the same header file:

- struct bpf_map_def is public, it is #included in the BPF .c source code
- struct bpf_map is an internal type for gobpf that should not be exposed. It is only used in `elf/elf.go`

So I would remove `bpf_map` and put it back in `elf/elf.go`